driver/src/contracts/base_contract.rs

```diff
@@ -0,0 +1,52 @@
+use std::env;
+
+use web3::contract::Contract;
+use web3::futures::Future;
+use web3::types::Address;
+
+use crate::error::DriverError;
+
+type Result<T> = std::result::Result<T, DriverError>;
+// TODO - Generalize the ContractImpl struct out from each of the two contract handlers
+
+#[allow(dead_code)] // event_loop needs to be retained to keep web3 connection open
+pub struct SmartContract {
```

Comment:
Maybe we should call this `ContractBase` or `BaseContract`? Smart Contract sounds a bit too powerful in my view.

Reply:
Yes! Good thinking!